job/job.go

```diff
@@ -77,8 +77,9 @@ type Job struct {
 	// Http client timeout for download in seconds
 	DownloadTimeout int `json:"download_timeout"`
 
-	// The User-Agent to set in download requests
-	UserAgent string `json:"user_agent"`
+	// The request headers provided by the user given as a JSON string
+	// This attribute is optional.
+	RequestHeaders string `json:"request_headers"`
```

Comment:
This should be a `map[string]string`. Is there any reason that we use the intermediate representation (string) instead?

Reply:
Right so, a reason would be so that we manually encode and decode the request headers payload without having to modify `SaveJob` method of `Storage`. But indeed what you propose is correct, i.e having `RequestHeaders` be a map. To that end, the `SaveJob` method has also been changed a bit in order to handle maps and serialize them correctly as json marshaled data, and i think it is much better now as this is much more flexible than having to manually encode/decode map payloads that could exist as attributes of `Job`.